LanguageFeatures/Extension-types/superinterfaces_of_extension_type_A08_t01.dart

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a compile-time error if two elements in the type list of
+/// the `implements` clause of an extension type declaration specifies the same
+/// type.
+///
+/// @description Checks that it is a compile-time error if two elements in the
+/// type list of the `implements` clause of an extension type declaration
+/// specifies the same type.
+/// @author sgrekhov22@gmail.com
+/// @issue 53790, 53791
+
+// SharedOptions=--enable-experiment=inline-class
+
+class A {}
+
+class B<T> {}
+
+extension type IntET(int _) implements int {}
+
+extension type ET<T extends num>(T _) implements num {}
+
+extension type ET1(A _) implements A, A {}
+//                                    ^
+// [analyzer] unspecified
+// [cfe] unspecified
+
+extension type ET2(B<int> _) implements B<int>, B<int> {}
+//                                              ^
+// [analyzer] unspecified
+// [cfe] unspecified
+
+extension type ET3(IntET _) implements IntET, int, IntET {}
+//                                               ^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+
+extension type ET4<T extends num>(B<T> _) implements B<T>, B<T> {}
+//                                                         ^
+// [analyzer] unspecified
+// [cfe] unspecified
+
+extension type ET5<T extends num>(T _) implements ET<T>, ET<T> {}
+//                                                       ^^
+// [analyzer] unspecified
+// [cfe] unspecified
+
```

Comment:
Perhaps add just one case to verify that there is no error if the two occurrences of the same type are not in the same list. E.g., in the current declaration and in a superinterface:

```dart
extension type ET6(A a) implements A {}
extension type ET7(A a) implement A, ET6 {}
```